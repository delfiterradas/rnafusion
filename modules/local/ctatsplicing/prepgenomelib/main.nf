process CTATSPLICING_PREPGENOMELIB {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://data.broadinstitute.org/Trinity/CTAT_SINGULARITY/CTAT-SPLICING/ctat_splicing.v0.0.2.simg' :
        'docker.io/trinityctat/ctat_splicing:0.0.2' }"

    input:
    tuple val(meta), path(genome_lib)
    path(cancer_intron_tsv)

    output:
    tuple val(meta), path(genome_lib, includeInputs:true), emit: reference
    path "versions.yml"                                  , emit: versions

    script:
    def VERSION = '0.0.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    /usr/local/src/CTAT-SPLICING/prep_genome_lib/ctat-splicing-lib-integration.py \\
        --cancer_introns_tsv cancer_introns.*.tsv.gz \\
        --genome_lib_dir $genome_lib

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ctat-splicing: $VERSION
    END_VERSIONS
    """

    stub:
    def VERSION = '0.0.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    touch $genome_lib/refGene.bed
    touch $genome_lib/refGene.sort.bed.gz
    touch $genome_lib/refGene.sort.bed.gz.tbi
    mkdir $genome_lib/cancer_splicing_lib
    touch $genome_lib/cancer_splicing_lib/cancer_splicing.idx

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ctat-splicing: $VERSION
    END_VERSIONS
    """
}
