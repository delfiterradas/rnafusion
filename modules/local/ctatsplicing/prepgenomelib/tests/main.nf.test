nextflow_process {

    name "Test Process CTATSPLICING_PREPGENOMELIB"
    script "../main.nf"
    process "CTATSPLICING_PREPGENOMELIB"

    setup {
        run("STARFUSION_BUILD") {
            script "../../../../nf-core/starfusion/build/main.nf"
            process {
                """
                input[0] = [
                    [ id:'minigenome fasta' ],
                    file("https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/master/minigenome.fa")
                ]
                input[1] = [
                    [ id:'minigenome gtf' ],
                    file("https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/master/minigenome.gtf")
                ]

                input [2] = file("https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/master/CTAT_HumanFusionLib.mini.dat.gz")
                input [3] = "homo_sapiens"
                input [4] = "3.8"
                input [5] = "37.4"                """
            }
        }
    }

    test("test") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = STARFUSION_BUILD.out.reference
                input[1] = [
                    file("https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/CANCER_SPLICING_LIB_SUPPLEMENT/cancer_introns.GRCh38.Jun232020.tsv.gz", checkIfExists:true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    path(process.out.reference[0][1]).resolve("AnnotFilterRule.pm"),
                    path(process.out.reference[0][1]).resolve("blast_pairs.dat.gz").exists(),
                    path(process.out.reference[0][1]).resolve("blast_pairs.idx").exists(),
                    path(process.out.reference[0][1]).resolve("__chkpts/annotfiltrule_cp.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/blast_pairs.idx.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/cp_gene_blast_pairs.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/cp_pfam_dat.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/cp_ref_annot_cdna.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/fusion_annot_lib.cp.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/_fusion_annot_lib.idx.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/index_pfam_hits.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/index_ref_annot_cdna.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/makeblastdb.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/mm2_genome_idx.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/mm2.splice_bed.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/_prot_info_db.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/ref_annot.cdsplus.dfam_masked.fa.cp.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/ref_annot.cdsplus.dfam_masked.fa.idx.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/ref_annot.gtf.gene_spans.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/ref_annot.gtf.mini.sortu.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/ref_annot.gtf.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/ref_genome_fai.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/ref_genome.fa.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/trans.blast.dat.cp.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/trans.blast.dat.index.ok"),
                    path(process.out.reference[0][1]).resolve("__chkpts/validate_ctat_genome_lib.ok"),
                    path(process.out.reference[0][1]).resolve("fusion_annot_lib.gz"),
                    path(process.out.reference[0][1]).resolve("fusion_annot_lib.idx").exists(),
                    path(process.out.reference[0][1]).resolve("pfam_domains.dbm").exists(),
                    path(process.out.reference[0][1]).resolve("PFAM.domtblout.dat.gz").exists(),
                    path(process.out.reference[0][1]).resolve("ref_annot.cdna.fa").exists(),
                    path(process.out.reference[0][1]).resolve("ref_annot.cdna.fa.idx").exists(),
                    path(process.out.reference[0][1]).resolve("ref_annot.cds").exists(),
                    path(process.out.reference[0][1]).resolve("ref_annot.cdsplus.fa").exists(),
                    path(process.out.reference[0][1]).resolve("ref_annot.cdsplus.fa.idx").exists(),
                    path(process.out.reference[0][1]).resolve("ref_annot.gtf"),
                    path(process.out.reference[0][1]).resolve("ref_annot.gtf.gene_spans").exists(),
                    path(process.out.reference[0][1]).resolve("ref_annot.gtf.mini.sortu"),
                    path(process.out.reference[0][1]).resolve("ref_annot.gtf.mm2.splice.bed"),
                    path(process.out.reference[0][1]).resolve("ref_annot.pep").exists(),
                    path(process.out.reference[0][1]).resolve("ref_annot.prot_info.dbm").exists(),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.fai"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.mm2"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.ndb"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.nhr"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.nin").exists(),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.njs").exists(),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.not"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.nsq"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.ntf"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.nto"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx").exists(),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/build.ok"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/chrLength.txt"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/chrNameLength.txt"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/chrName.txt"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/chrStart.txt"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/exonGeTrInfo.tab"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/exonInfo.tab"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/geneInfo.tab"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/Genome"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/genomeParameters.txt").exists(),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/Log.out").exists(),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/SA"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/SAindex"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/sjdbInfo.txt"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/sjdbList.fromGTF.out.tab"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/sjdbList.out.tab"),
                    path(process.out.reference[0][1]).resolve("ref_genome.fa.star.idx/transcriptInfo.tab"),
                    path(process.out.reference[0][1]).resolve("trans.blast.align_coords.align_coords.dat"),
                    path(process.out.reference[0][1]).resolve("trans.blast.align_coords.align_coords.dbm").exists(),
                    path(process.out.reference[0][1]).resolve("trans.blast.dat.gz"),
                    path(process.out.reference[0][1]).resolve("refGene.sort.bed.gz"),
                    path(process.out.reference[0][1]).resolve("refGene.sort.bed.gz.tbi"),
                    path(process.out.reference[0][1]).resolve("cancer_splicing_lib/cancer_splicing.idx").exists(),
                    process.out.version
                ).match()
            }
            )
        }
    }

    test("test - stub") {

        options "-stub"

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = STARFUSION_BUILD.out.reference
                input[1] = [
                    file("https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/CANCER_SPLICING_LIB_SUPPLEMENT/cancer_introns.GRCh38.Jun232020.tsv.gz", checkIfExists:true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.findAll { key, value -> !key.isNumber() }).match() }
            )
        }
    }
}
