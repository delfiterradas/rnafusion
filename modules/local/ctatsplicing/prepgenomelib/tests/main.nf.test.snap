{
    "test": {
        "content": [
            {
                "reference": [
                    [
                        {
                            "id": "minigenome fasta"
                        },
                        [
                            "AnnotFilterRule.pm:md5,f94966013cd0df9624a6dda0b75fefa0",
                            "PFAM.domtblout.dat.gz:md5,b8cf97310b4af1275237febab723d16c",
                            [
                                "_fusion_annot_lib.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "_prot_info_db.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "annotfiltrule_cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "blast_pairs.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "cp_gene_blast_pairs.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "cp_pfam_dat.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "cp_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "fusion_annot_lib.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "index_pfam_hits.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "index_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "makeblastdb.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "mm2.splice_bed.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "mm2_genome_idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.cdsplus.dfam_masked.fa.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.cdsplus.dfam_masked.fa.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.gtf.gene_spans.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.gtf.mini.sortu.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.gtf.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_genome.fa.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_genome_fai.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "trans.blast.dat.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "trans.blast.dat.index.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "validate_ctat_genome_lib.ok:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "blast_pairs.dat.gz:md5,31ec48d5116f44b12fdd3f47155241dc",
                            "blast_pairs.idx:md5,0264ed1a28c50370e598cc8bd874dce0",
                            [
                                "cancer_splicing.idx:md5,fcec006060ebd0bfb1b0cf38c9f78116"
                            ],
                            "fusion_annot_lib.gz:md5,23d82a5da81f91ca4e1ecd6481992a12",
                            "fusion_annot_lib.idx:md5,21bc9b75123be07089e86023285b67f8",
                            "pfam_domains.dbm:md5,68a00826b188aa577237a701e6f18e8c",
                            "refGene.bed:md5,a14d9396c44c5fc0cd6b4d3194e8902d",
                            "refGene.sort.bed.gz:md5,01bf4093e3e9dcad8ed3a0f45a381c95",
                            "refGene.sort.bed.gz.tbi:md5,6dd8cd94a5cd10699fe8445a70c5b232",
                            "ref_annot.cdna.fa:md5,4907484edddb1a52d1ef5acb814fa709",
                            "ref_annot.cdna.fa.idx:md5,8f5c6a6c2ffec0188fac2483f806aabc",
                            "ref_annot.cds:md5,de204070512c909b351c5876133f4f54",
                            "ref_annot.cdsplus.fa:md5,180f4e5d277f2d5fc4bb092c11eaaf68",
                            "ref_annot.cdsplus.fa.idx:md5,be8bb33932b6c2f82213a0622241ad0f",
                            "ref_annot.gtf:md5,5ce8afe99ef3940a877a04caeacf9181",
                            "ref_annot.gtf.gene_spans:md5,4884ab6d453096b48bf85595299d9183",
                            "ref_annot.gtf.mini.sortu:md5,1d29ccecdbb7b40a99c84a02d6c2c1be",
                            "ref_annot.gtf.mm2.splice.bed:md5,340585ea1843bf06bf555575ddecf28c",
                            "ref_annot.pep:md5,22f05c0bf182e2645f0a01c4fb45e908",
                            "ref_annot.prot_info.dbm:md5,218a49ac07f0db5fba0b1b48e6b12343",
                            "ref_genome.fa:md5,ad699c56ed38566c7d3e9579486b1706",
                            "ref_genome.fa.fai:md5,e3f74a27219b33ae80fd5de5cbeaf32b",
                            "ref_genome.fa.mm2:md5,ce50979ea284748eb9f84ae88cfd930e",
                            "ref_genome.fa.ndb:md5,6ea574753b557610f62f6e4ab79e19f5",
                            "ref_genome.fa.nhr:md5,50f28dae71683c4394bfaf94a1ef4392",
                            "ref_genome.fa.nin:md5,3c4114ca4fb5dd0dc2095f4cf5f7b844",
                            "ref_genome.fa.njs:md5,39b423551f658bb0cb5032ac00bd74eb",
                            "ref_genome.fa.not:md5,1e53e9d08f1d23af0299cfa87478a7bb",
                            "ref_genome.fa.nsq:md5,d2361e7871ce4cf51181c112a48f191b",
                            "ref_genome.fa.ntf:md5,de1250813f0c7affc6d12dac9d0fb6bb",
                            "ref_genome.fa.nto:md5,33cdeccccebe80329f1fdbee7f5874cb",
                            [
                                "Genome:md5,9e3efdd0901cabb5a2d589664a63b372",
                                "Log.out:md5,5310b905153fa779043520ec3fc7ca3b",
                                "SA:md5,7dd9083264be9c6a2194d990bc10d237",
                                "SAindex:md5,ac4711df685109e04356db9e9cb9fb7f",
                                "build.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "chrLength.txt:md5,e02cd536b7281b894246863b160d5d06",
                                "chrName.txt:md5,f4d0d6595f423084e6b9472e40dfe6e8",
                                "chrNameLength.txt:md5,07a67d7ac441d7d30d80840b0927e717",
                                "chrStart.txt:md5,e2031239a74fe5ee9051e9364e4f608a",
                                "exonGeTrInfo.tab:md5,3c35618d07a8e35a0f9108699fcdda42",
                                "exonInfo.tab:md5,bcbb3f32fa31fe504cc737f337ad341c",
                                "geneInfo.tab:md5,db5db4b6e003904e9908fce7c05f0125",
                                "genomeParameters.txt:md5,9664cf5658a941c5d4cf9f5bb5e5cb4f",
                                "sjdbInfo.txt:md5,e4cc1bbf8bd687cfc3d7c2c702e6def7",
                                "sjdbList.fromGTF.out.tab:md5,8f3e8604b00d4067e4eb80aa476a8113",
                                "sjdbList.out.tab:md5,5d78dd49d5db24ca2c056b7ebe5c2059",
                                "transcriptInfo.tab:md5,b758c0ccaddcf0453bab5905b3cec4a2"
                            ],
                            "trans.blast.align_coords.align_coords.dat:md5,9f6b7a75aea03a9671190be25ecdd4c2",
                            "trans.blast.align_coords.align_coords.dbm:md5,4ae2e5e8d141bd2e066da74ffb60403b",
                            "trans.blast.dat.gz:md5,85ba5ea96c566f751ad83a3e4b8ab128"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8175cd19f5d5b8b2214cd948120d7524"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-16T09:41:16.195052"
    },
    "test - stub": {
        "content": [
            {
                "reference": [
                    [
                        {
                            "id": "minigenome fasta"
                        },
                        [
                            "AnnotFilterRule.pm:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "PFAM.domtblout.dat.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            [
                                "_fusion_annot_lib.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "_prot_info_db.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "annotfiltrule_cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "blast_pairs.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "cp_gene_blast_pairs.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "cp_pfam_dat.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "cp_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "fusion_annot_lib.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "index_pfam_hits.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "index_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "makeblastdb.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "mm2.splice_bed.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "mm2_genome_idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.cdsplus.dfam_masked.fa.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.cdsplus.dfam_masked.fa.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.gtf.gene_spans.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.gtf.mini.sortu.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_annot.gtf.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_genome.fa.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "ref_genome_fai.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "trans.blast.dat.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "trans.blast.dat.index.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "validate_ctat_genome_lib.ok:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "blast_pairs.dat.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "blast_pairs.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                            [
                                "cancer_splicing.idx:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "fusion_annot_lib.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "fusion_annot_lib.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "pfam_domains.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "refGene.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "refGene.sort.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "refGene.sort.bed.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.cdna.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.cdna.fa.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.cds:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.cdsplus.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.cdsplus.fa.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf.gene_spans:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf.mini.sortu:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf.mm2.splice.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.pep:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.prot_info.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.mm2:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.ndb:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.nhr:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.nin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.njs:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.not:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.nsq:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.ntf:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.nto:md5,d41d8cd98f00b204e9800998ecf8427e",
                            [
                                "Genome:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "Log.out:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "SA:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "SAindex:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "build.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "chrLength.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "chrName.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "chrNameLength.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "chrStart.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "exonGeTrInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "exonInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "geneInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "genomeParameters.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sjdbInfo.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sjdbList.fromGTF.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "sjdbList.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "transcriptInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "trans.blast.align_coords.align_coords.dat:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "trans.blast.align_coords.align_coords.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "trans.blast.dat.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8175cd19f5d5b8b2214cd948120d7524"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-19T09:17:05.045966772"
    }
}